### PR TITLE
feat: lifei/configure system

### PR DESCRIPTION
**Pull request by lifeizhou-ap**
**What**
- Ask user to add/remove system with url via sub command `goose system add $url` and `goose system remove $url`
- call `${url}/fetch_name` to get the system name
   -  Questions:
      -  We may have a single url to fetch name, descriptions?
      - This could be also be moved the goose core code?
- Save the system location(url) and system name in the config file for all the profiles
- Created separate project to start the stub system server.  

**Screenshots**
Add system
Screenshot 2024-11-19 at 5 38 48 PM

Remove system

Screenshot 2024-11-19 at 5 39 48 PM

**Note**
Maybe we can also enable adding multiple urls in the same subcommand in the future

### CODE REVIEW

**Comment by michaelneale** [comment]
nice - not been over the code too closely but looks good - should this also have it in goose-server so it can also register systems? I like this!

File: crates/goose-cli/src/commands/configure.rs [hunk 1]
```diff
@@ -38,6 +44,46 @@ pub async fn handle_configure(provided_profile_name: Option<String>) -> Result<(
     Ok(())
 }
 
+async fn set_systems(additional_systems: &mut Vec<AdditionalSystem>) -> Result<(), Box<dyn Error>> {
```
Comment (baxen) [inline crates/goose-cli/src/commands/configure.rs]: i'd probably remove this from the configure command entirely i think? you've figured out how here but its kind of awkward to implement a dialogue looping around additions

I'd suggest at the end of configure we say "we've added the developer system by default, to add or remove systems, use \n goose system {add|remove} systemurl"

Comment (baxen) [reply crates/goose-cli/src/commands/configure.rs]: then we'd need a new subcommand

**Comment by lifeizhou-ap** [comment]
> nice - not been over the code too closely but looks good - should this also have it in goose-server so it can also register systems? I like this!

Thanks! @michaelneale 

I had asked  @baxen about this while he was reviewing this PR.  Bradley suggested "let's wait a min i think, since it might change with MCP".  We can adapt this after the change with MCP

**Pull request merged**